aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/crypto/ContentCryptoScheme.java

```diff
@@ -177,10 +177,10 @@ CipherLite createCipherLite(SecretKey cek, byte[] iv, int cipherMode,
         String specificProvider = getSpecificCipherProvider();
         Cipher cipher;
         try {
-            if (specificProvider != null) { // use the specific provider if defined
-                    cipher = Cipher.getInstance(getCipherAlgorithm(), specificProvider);
-            } else if (securityProvider != null) { // use the one optionally specified in the input
+            if (securityProvider != null) { // use the one optionally specified in the input
```

Comment:
Hmm I'm a little apprehensive this change in precedence since `securityProvider` is clearly documented as only used if `specificProvider` is null:

>      * @param securityProvider
>      *            optional security provider to be used but only if there is no
>      *            specific provider defined for the specified scheme.

I think I'd be more confortable having this be opt-in behavior.

Reply:
Cool, I'll plumb that through.